pandas/core/arrays/integer.py

```diff
@@ -553,6 +553,38 @@ def cmp_method(self, other):
         name = f"__{op.__name__}__"
         return set_function_name(cmp_method, name, cls)
 
+    def _accumulate(self, name: str, skipna: bool = True, **kwargs) -> "IntegerArray":
+        data = self._data
+        mask = self._mask
+
+        cum_function, fill_value = {
+            "cumprod": (np.cumprod, 1),
+            "cummax": (np.maximum.accumulate, data.min()),
+            "cumsum": (np.cumsum, 0),
+            "cummin": (np.minimum.accumulate, data.max()),
+        }[name]
```

Comment:
this would raise KeyError on an invalid function, but that is not a friendly message to the user, you can either catch it or (do an in on the dict first) and provide a nice ValueError message

Reply:
done